amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/holds/HoldsChildrenRelation.java

```diff
@@ -0,0 +1,182 @@
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.transaction.TransactionService;
+import org.springframework.beans.factory.InitializingBean;
+
+/**
+ * Hold children relation
+ *
+ * @author Damian Ujma
+ */
+@RelationshipResource(name = "children", entityResource = HoldsEntityResource.class, title = "Children of a hold")
+public class HoldsChildrenRelation implements
+    RelationshipResourceAction.Create<HoldChild>,
+    RelationshipResourceAction.Read<HoldChild>,
+    RelationshipResourceAction.Delete,
+    InitializingBean
+{
+    private HoldService holdService;
+    private FilePlanComponentsApiUtils apiUtils;
+    private ApiNodesModelFactory nodesModelFactory;
+    private TransactionService transactionService;
+    private FileFolderService fileFolderService;
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        mandatory("holdService", holdService);
+        mandatory("apiUtils", apiUtils);
+        mandatory("nodesModelFactory", nodesModelFactory);
+        mandatory("transactionService", transactionService);
+        mandatory("fileFolderService", fileFolderService);
+    }
+
+    @Override
+    @WebApiDescription(title = "Add one (or more) children as children of a hold identified by 'holdId'")
+    public List<HoldChild> create(String holdId, List<HoldChild> children, Parameters parameters)
+    {
+        // validate parameters
+        checkNotBlank("holdId", holdId);
+        mandatory("children", children);
+        mandatory("parameters", parameters);
+
+        NodeRef parentNodeRef = apiUtils.lookupAndValidateNodeType(holdId, RecordsManagementModel.TYPE_HOLD);
+
+        RetryingTransactionCallback<List<NodeRef>> callback = () -> {
+            List<NodeRef> createdNodes = children.stream()
+                .map(holdChild -> new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, holdChild.getId()))
+                .collect(Collectors.toList());
+            holdService.addToHold(parentNodeRef, createdNodes);
+            return createdNodes;
+        };
+
+        List<NodeRef> nodeInfos = transactionService.getRetryingTransactionHelper()
+            .doInTransaction(callback, false, true);
+
+        return nodeInfos.stream()
+            .map(nodeRef -> new HoldChild(nodeRef.getId()))
+            .collect(Collectors.toCollection(LinkedList::new));
+    }
+
+    @Override
+    @WebApiDescription(title = "Return a paged list of hold container children for the container identified by 'holdContainerId'")
+    public CollectionWithPagingInfo<HoldChild> readAll(String holdId, Parameters parameters)
+    {
+        checkNotBlank("holdId", holdId);
+        mandatory("parameters", parameters);
+
+        Paging paging = parameters.getPaging();
+        NodeRef parentNodeRef = apiUtils.lookupAndValidateNodeType(holdId, RecordsManagementModel.TYPE_HOLD);
+        List<NodeRef> holds = holdService.getHeld(parentNodeRef);
```

Comment:
This is a list of held nodes, not a list of holds. I would call it something like `heldNodes` or `children`.

Reply:
Changed to `children`.